Source/WebCore/platform/sql/SQLiteFileSystem.cpp

```diff
@@ -157,14 +157,11 @@ String SQLiteFileSystem::computeHashForFileName(StringView fileName)
     auto digest = cryptoDigest->computeHash();
     
     // Convert digest to hex.
-    char* start = 0;
+    std::span<char> start;
     unsigned digestLength = digest.size();
     CString result = CString::newUninitialized(digestLength * 2, start);
-    char* buffer = start;
-    for (size_t i = 0; i < digestLength; ++i) {
-        snprintf(buffer, 3, "%02X", digest.at(i));
-        buffer += 2;
-    }
+    for (size_t i = 0; i < digestLength; ++i)
+        snprintf(start.subspan(i * 2).data(), 3, "%02X", digest.at(i));
```

Comment:
This copied and pasted code should probably share a function with the one from `SHA1.cpp`. The same mistakes are here. But also here seems unnecessary to use a `CString` and turn it into a `String`. More efficient to make a Latin-1 `String` directly. So maybe need a separate `String` and `CString` version?